src/pages/index.js

```diff
@@ -72,50 +88,166 @@ newCardPopup.setEventListeners();
 
+const avatarPopup = new PopupWithForm("#avatar__modal", handleAvatarUpdate);
+avatarPopup.setEventListeners();
+
+const deleteConfirm = new PopupWithConfirm("#delete__modal", handleCardDelete);
+deleteConfirm.setEventListeners();
+
+const api = new Api({
+  baseURL: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "4792ec92-cf1c-45a6-8740-0f5d63585faa",
+    "Content-type": "application/json",
+  },
+});
+
 //////////////////////////////////////////// FUNCTIONS //////////////////////////////////////////////////////
 
 function handleImageClick(name, link) {
   popupImage.openPopup({ name, link });
 }
 
 function handleProfileModalSubmit(profileData) {
+  profileModalSubmitButton.textContent = "Saving...";
   console.log("profileData:", profileData);
   const name = profileData.title; //set name var to work with userInfo class
   const job = profileData.description; //set job var to work with userInfo class
   console.log("name:", name, "job:", job); //debug log
   user.setUserInfo({ name, job }); //set user information
+  api.updateUserInfo(profileData.title, profileData.description);
   const updatedUserInfo = user.getUserInfo(); //get updated information and store in a var
   console.log(updatedUserInfo); // debug log
   profilePopup.closePopup(); //close popup
+
+  setTimeout(() => {
+    profileModalSubmitButton.textContent = "Save";
+  }, 1000);
+}
+
+function handleAvatarUpdate(link) {
+  const url = link.url; //extract url from link object and set it to url value
+
+  avatarModalSubmitButton.textContent = "Saving...";
+
+  api
+    .updateProfilePic(url)
+    .then(() => {
+      user.setUserAvatar(url);
+    })
+    .catch((err) => {
+      console.error(`Error caught in HandleAvatarUpdate function ${err}`);
+    });
+
+  setTimeout(() => {
+    avatarModalSubmitButton.textContent = "Save";
+  }, 1000);
+  avatarPopup.closePopup();
 }
 
 function handleAddCardFormSubmit(newCardData) {
-  const name = newCardData.title; //set card name
-  const alt = newCardData.title; //set card alt text
-  const link = newCardData.url; //set card image link
-  console.log(name, alt, link, newCardData); //debug log
-
-  renderCard({ name, link, alt }); //call renderCard to append card to page
-  newCardPopup.closePopup(); //close popup
-  addNewCardValidator.resetForm(); //reset form and disable submit button
+  addNewCardSubmitButton.textContent = "Adding...";
+  api
+    .createNewCard(newCardData.title, newCardData.url)
+    .then((res) => {
+      /* console.log(res); */
+      renderCard(res);
+      newCardPopup.closePopup(); //close popup
+      addNewCardValidator.resetForm(); //reset form and disable submit button
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+
+  setTimeout(() => {
+    addNewCardSubmitButton.textContent = "Add";
+  }, 1000);
+}
+
+function handleCardLikes(cardData) {
+  /*   console.log(cardData);
+  console.log(cardData._like); */
+
+  if (!cardData._like) {
```

Comment:
`_like`

Please, delete the **low dash** (`underscore`) from methods/fields if you want to use them **outside** a class because the `low dash` shows a **private** field which is used only inside the class.

Reply:
Please, delete the **low dash** (`underscore`) from methods/fields if you want to use them **outside** a class because the `low dash` shows a **private** field which is used only inside the class.